packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderItem.tsx

```diff
@@ -0,0 +1,184 @@
+import * as React from 'react';
+import { COLUMN_HEADER_CLICK } from '../../constants/eventsConstants';
+import { ColDef, NUMBER_COLUMN_TYPE } from '../../models/colDef/index';
+import { GridOptions } from '../../models/gridOptions';
+import { ColParams } from '../../models/params/colParams';
+import { SortDirection } from '../../models/sortModel';
+import { ApiContext } from '../api-context';
+import { HEADER_CELL_CSS_CLASS } from '../../constants/cssClassesConstants';
+import { classnames } from '../../utils/index';
+import { ColumnHeaderSortIcon } from './ColumnHeaderSortIcon';
+import { ColumnHeaderTitle } from './ColumnHeaderTitle';
+import { ColumnHeaderSeparator } from './ColumnHeaderSeparator';
+import { ColumnHeaderMenuIcon } from './ColumnHeaderMenuIcon';
+import { ColumnHeaderFilterIcon } from './ColumnHeaderFilterIcon';
+
+interface ColumnHeaderItemProps {
+  colIndex: number;
+  column: ColDef;
+  isDragging: boolean;
+  isResizing: boolean;
+  sortDirection: SortDirection;
+  sortIndex?: number;
+  options: GridOptions;
+  separatorProps: React.HTMLAttributes<HTMLDivElement>;
+  filterItemsCounter?: number;
+}
+
+export const ColumnHeaderItem = ({
+  column,
+  colIndex,
+  isDragging,
+  isResizing,
+  separatorProps,
+  sortDirection,
+  sortIndex,
+  options,
+  filterItemsCounter,
+}: ColumnHeaderItemProps) => {
+  const apiRef = React.useContext(ApiContext);
+  const {
+    disableColumnReorder,
+    showColumnRightBorder,
+    disableColumnResize,
+    disableColumnMenu,
+  } = options;
+  const isColumnSorted = column.sortDirection != null;
+  // todo refactor to a prop on col isNumeric or ?? ie: coltype===price wont work
+  const isColumnNumeric = column.type === NUMBER_COLUMN_TYPE;
+
+  let headerComponent: React.ReactElement | null = null;
+  if (column.renderHeader) {
+    headerComponent = column.renderHeader({
+      api: apiRef!.current!,
+      colDef: column,
+      colIndex,
+      field: column.field,
+    });
+  }
+
+  const onDragStart = React.useCallback(
+    (event) => apiRef!.current.onColItemDragStart(column, event.currentTarget),
+    [apiRef, column],
+  );
+  const onDragEnter = React.useCallback((event) => apiRef!.current.onColItemDragEnter(event), [
+    apiRef,
+  ]);
+  const onDragOver = React.useCallback(
+    (event) =>
+      apiRef!.current.onColItemDragOver(column, {
+        x: event.clientX,
+        y: event.clientY,
+      }),
+    [apiRef, column],
+  );
+  const onHeaderTitleClick = React.useCallback(() => {
+    const colHeaderParams: ColParams = {
+      field: column.field,
+      colDef: column,
+      colIndex,
+      api: apiRef!.current,
+    };
+    apiRef!.current.publishEvent(COLUMN_HEADER_CLICK, colHeaderParams);
+  }, [apiRef, colIndex, column]);
+
+  const cssClasses = React.useMemo(
```

Comment:
Did you measure a difference in performance?

```suggestion
  const cssClasses = 
```